kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/KieServerImpl.java

```diff
@@ -630,6 +625,64 @@ private void storeScannerState(String containerId, KieScannerResource scannerSta
         repository.store(KieServerEnvironment.getServerId(), currentState);
     }
 
+    private <T> void storeServerState(String containerId, T resource, Object... extraRes) {
```

Comment:
This approach looks quite error prone to me.
The functionality is split into 2 parts (here and in specific implementation methods), making the code more confusing.
Switching functionality based on class name increases risk of bug in case a new functionality is added as this doesn't indicate in compile time that any change is required in this method.
My suggestion is to return functionality back to classes and change storeServerState implementation to something like:
```
    private void storeServerState(Consumer<KieServerState> kieServerStateConsumer) {
        KieServerState currentState = repository.load(KieServerEnvironment.getServerId());
        kieServerStateConsumer.accept(currentState);
        repository.store(KieServerEnvironment.getServerId(), currentState);
    }
```
In code it can look like:
```
                            container.setStatus(KieContainerStatus.STARTED);
                            storeServerState(state -> {
                                state.getContainers().add(container);
                            });
```

Reply:
Refactored as per suggestion. For now, leave the original implementation as deprecated methods. Once we complete final testes, we can remove them.